braze-inc/braze-docs
# bdocs: script for auto-generating redirects

### Why are you making this change? (required)

This PR contains a new script for auto-generating the redirect file. Here's the [requirements doc](https://docs.google.com/document/d/13sBH2IRzjpdugns5042RS6feVk2cXCawvqOnYReDZ4A/edit?tab=t.0#heading=h.ugnkiqrhq271) for more info.

### TESTING THE SCRIPT

1. Rename one or more files in `_docs`.
2. Run: `git add . && git commit -m "test"`
3. Run: `./bdocs mredirects`
4. Verify redirect was added to `assets/js/broken_redirect_list.js`

### TODO

- [x] inform users that they need to make a `git commit` before they can generate redirects.
- [x] inform users if command was successful but no redirects were added.
- [x] ensure duplicates are removed
- [x] cleanup comments/general structure (if applicable).

### Related PRs, issues, or features (optional)

- N/A

### Feature release date (optional)

- N/A

### Contributor checklist

- [x] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.
